### PR TITLE
FOGL-973 Add support for specific timezones in query interface

### PR DESCRIPTION
**Pull request by MarkRiddoch**

### CODE REVIEW

File: C/plugins/storage/postgres/connection.cpp [hunk 1]
```diff
@@ -144,7 +144,7 @@ SQLBuffer	jsonConstraints;	// Extra constraints to add to where clause
 							{
 								sql.append((*itr)["column"].GetString());
 								sql.append(" AT TIME ZONE '");
-								sql.append((*itr)["timezone"]);
+								sql.append((*itr)["timezone"].GetString());
```
Comment (pintomax) [inline C/plugins/storage/postgres/connection.cpp]: We might check the provided timezone with IsString() in order to prevent corner case issues.

**Comment by sstefano69** [comment]
@Mark, 

I saw this error  in the syslog : 

`Jan` 22 22:02:15 foglamp FogLAMP Storage[23381]: PostgreSQL storage plugin raising error: ERROR:  `syntax` error at or near "SORT"#012LINE 1: ...ME ZONE 'UTC' FROM foglamp.readings WHERE id >= 1 `SORT` BY id...#012                                                             ^

it seems the alias are no more there.

File: C/plugins/storage/postgres/connection.cpp [hunk 2]
```diff
@@ -749,7 +772,7 @@ bool Connection::fetchReadings(unsigned long id, unsigned int blksize, std::stri
 char	sqlbuffer[200];
 
 	snprintf(sqlbuffer, sizeof(sqlbuffer),
-		"SELECT id, asset_code, read_key, reading, user_ts AT TIME ZONE 'UTC', ts AT TIME ZONE 'UTC' FROM foglamp.readings WHERE id >= %ld LIMIT %d;", id, blksize);
+		"SELECT id, asset_code, read_key, reading, user_ts AT TIME ZONE 'UTC', ts AT TIME ZONE 'UTC' FROM foglamp.readings WHERE id >= %ld SORT BY id LIMIT %d;", id, blksize);
```
Comment (sstefano69) [inline C/plugins/storage/postgres/connection.cpp]: - I think 'SORT' should be 'ORDER'
- we should add 'as user_ts' and 'as ts'

**Comment by sstefano69** [comment]
the call to - statistics_history = self._storage.query_tbl_with_payload('statistics_history', payload) 
is still returning a timestamp having the timezone, sample :'2018-01-22 22:34:55.273496+05:30'

File: C/plugins/storage/postgres/connection.cpp [hunk 3]
```diff
@@ -749,7 +772,7 @@ bool Connection::fetchReadings(unsigned long id, unsigned int blksize, std::stri
 char	sqlbuffer[200];
 
 	snprintf(sqlbuffer, sizeof(sqlbuffer),
-		"SELECT id, asset_code, read_key, reading, user_ts AT TIME ZONE 'UTC' as \"user_ts\", ts AT TIME ZONE 'UTC' as \"ts\" FROM foglamp.readings WHERE id >= %ld LIMIT %d;", id, blksize);
+		"SELECT id, asset_code, read_key, reading, user_ts AT TIME ZONE 'UTC' as \"user_ts\", ts AT TIME ZONE 'UTC' as '|ts\" FROM foglamp.readings WHERE id >= %ld ORDER BY id LIMIT %d;", id, blksize);
```
Comment (sstefano69) [inline C/plugins/storage/postgres/connection.cpp]: I got the error  : 
`Jan 23 16:07:13 foglamp FogLAMP Storage[26252]: PostgreSQL storage plugin raising error: ERROR:  unterminated quoted string at or near "'|ts" FROM foglamp.readings WHERE id >= 1 ORDER BY id LIMIT 5;"#012LINE 1: ...ZONE 'UTC' as "user_ts", ts AT TIME ZONE 'UTC' as '|ts" FROM...#012                                                             ^
`
it should be the '|ts\"

**Pull request merged**